### PR TITLE
Fix per-project configuration behaviour

### PR DESCRIPTION
**Pull request by rarguelloF**
Per-project settings should take precedence over user settings, just as stated in the docs:

> Note that package settings from Settings panel will be ignored if a configuration file was found.

This PR fix this behaviour and ignores user settings if finds a local configuration file.

### CODE REVIEW

**Comment by Arcanemagus** [comment]
Huh, I was under the impression (from that description) that specifying a config file caused `flake8` to ignore those flags, if the flags are overriding the config file (which makes sense) then I agree with you that this should be changed.

**Comment by rarguelloF** [comment]
Yes, `flake8` doesn't ignore those flags with the `--config` flag :(

Btw, I've just realized my solution has a problem. I should check also the config file exists, not only if the parameter is specified (I'll update my solution soon).

File: lib/main.coffee [hunk 1]
```diff
@@ -188,6 +188,18 @@ module.exports =
         return p
     return execPath
 
+  getExistingConfigFiles: (configFiles) ->
+    return configFiles
+      .split(/[ ,]+/)
+      .map (configFile) -> path.join(atom.project.getPaths()[0], configFile)
```
Comment (Arcanemagus) [inline lib/main.coffee]: Note that this will fail if the file isn't opened in a project.

**Comment by rarguelloF** [comment]
I updated my solution. Take it a look and tell me if you see something wrong :D

File: lib/main.coffee [hunk 2]
```diff
@@ -188,6 +188,32 @@ module.exports =
         return p
     return execPath
 
+  getProjectPath: (lintedFile) ->
```
Comment (Arcanemagus) [inline lib/main.coffee]: This whole function can be replaced with:
```js
atom.project.relativizePath(lintedFile)[0]
```

See [here](https://atom.io/docs/api/v1.8.0/Project#instance-relativizePath) for the documentation.

Comment (Arcanemagus) [inline lib/main.coffee]: Note that you need to handle when that returns `null` (the file not being opened in a project).

File: lib/main.coffee [hunk 3]
```diff
@@ -188,6 +188,32 @@ module.exports =
...
+
+    projectName = projectNames
+      .find((projectName) -> lintedFile.indexOf(projectName) > -1)
+
+    return projectPaths[projectNames.indexOf(projectName)]
+
+
+  getExistingConfigFiles: (configFiles, projectPath) ->
+    return configFiles
+      .split(/[ ,]+/)
```
Comment (Arcanemagus) [inline lib/main.coffee]: The config option mentions this is only the file name, but this seems to turn it into an array of _potential_ file names?

File: lib/main.coffee [hunk 4]
```diff
@@ -188,6 +188,32 @@ module.exports =
...
+      .map (configFile) -> path.join(projectPath, configFile)
```
Comment (Arcanemagus) [inline lib/main.coffee]: Check out the `helpers.findCached()` function, you pass it a directory and a (an array of) name(s) and it will find them for you.

**Comment by Arcanemagus** [comment]
@rarguelloF Are you interested in addressing the review comments?

**Comment by rarguelloF** [comment]
Yes I'm interested, I'll review your comments this week.

And sorry for disappearing so long, I've had some extremely busy weeks 😢 

Thanks for your patience!! 😄

**Comment by Arcanemagus** [comment]
No worries, it took me that long to check back 😛.

File: lib/main.coffee (hunk 3 above)
Comment (rarguelloF) [inline lib/main.coffee]: Yes, I'll explain myself: this function checks if any of the config files specified on the Atom configuration exists within the project or not. Later, when I push the `--config` parameter, I use the first config file that exists. If none of the configured filenames exists, the `--config` parameter is ignored.

This is the 3d check I do before pushing or not the `--config` parameter

```coffeescript
if (
  (projectConfigFile = atom.config.get('linter-flake8.projectConfigFile')) and
  (projectPath = atom.project.relativizePath(filePath)[0]) and
  (existingConfigFiles = @getExistingConfigFiles(projectConfigFile, projectPath)).length > 0
)
```

Comment (rarguelloF) [inline lib/main.coffee]: BTW I realized `helpers.findCached` already does this, so I'm using that function instead 💪

**Comment by rarguelloF** [comment]
@Arcanemagus I've already addressed the issues. Please tell me if you want me to change something else :D

**Comment by Arcanemagus** [comment]
LGTM, merging!

**Comment by Arcanemagus** [comment]
Published in v1.13.3 🎉

**Pull request merged**